src/main/kotlin/no/nav/syfo/client/OppgaveClient.kt

```diff
@@ -23,33 +28,73 @@ class OppgaveClient(
 ) {
     suspend fun opprettOppgave(opprettOppgave: OpprettOppgave, msgId: String):
             OpprettOppgaveResponse = retry("create_oppgave") {
-        httpClient.post<OpprettOppgaveResponse>(url) {
+
+        log.info("Oppretter oppgave {} ", opprettOppgave)
+
+        val httpResponse = httpClient.post<HttpStatement>(url) {
             contentType(ContentType.Application.Json)
             val oidcToken = oidcClient.oidcToken()
             header("Authorization", "Bearer ${oidcToken.access_token}")
             header("X-Correlation-ID", msgId)
             body = opprettOppgave
+        }.execute()
+
+        when (httpResponse.status) {
+            HttpStatusCode.Created -> {
+                log.info("OppgaveClient opprettOppgave svarte 201 CREATED")
+                httpResponse.call.response.receive<OpprettOppgaveResponse>()
+            }
+            else -> {
+                log.error("OppgaveClient opprettOppgave kastet feil {} ved opprettelse av oppgave", httpResponse.status)
+                throw RuntimeException("OppgaveClient opprettOppgave kastet feil $httpResponse.status")
+            }
         }
     }
 
     suspend fun ferdigStillOppgave(ferdigstilloppgave: FerdigStillOppgave, msgId: String):
             OpprettOppgaveResponse = retry("ferdigstill_oppgave") {
-        httpClient.patch<OpprettOppgaveResponse>(url + "/" + ferdigstilloppgave.id) {
+
+        log.info("Ferdigstiller oppgave {} ", ferdigstilloppgave)
+
+        val httpResponse = httpClient.patch<HttpStatement>(url + "/" + ferdigstilloppgave.id) {
             contentType(ContentType.Application.Json)
             val oidcToken = oidcClient.oidcToken()
             header("Authorization", "Bearer ${oidcToken.access_token}")
             header("X-Correlation-ID", msgId)
             body = ferdigstilloppgave
+        }.execute()
+
+        when (httpResponse.status) {
+            HttpStatusCode.OK -> {
+                httpResponse.call.response.receive<OpprettOppgaveResponse>()
+            }
+            else -> {
+                log.error("OppgaveClient ferdigStillOppgave kastet feil {} ved ferdigstilling av oppgave", httpResponse.status)
```

Comment:
Kan vi ha med oppgaveid i errorloggen? Samme gjelder error-loggen for henting av oppgave :) Da er det litt lettere å se hvilken oppgave som trigget feil.

Reply:
Jepp!